src/api/rpc_api/author.rs

```diff
@@ -106,13 +106,21 @@ pub trait SubmitAndWatch {
 
 	/// Submit an extrinsic and watch it until the desired status
 	/// is reached, if no error is encountered previously.
+	/// The events are not fetched. So no events are listed in the report.
+	/// To fetch the triggered events, please use submit_and_watch_extrinsic_until.
 	/// Upon success, a report containing the following information is returned:
 	/// - extrinsic hash
 	/// - if watched until at least `InBlock`:
 	///   hash of the block the extrinsic was included in
 	/// - last known extrinsic (transaction) status
+	/// - no events
 	/// This method is blocking.
-	async fn submit_and_watch_extrinsic_until<Address, Call, Signature, SignedExtra>(
+	async fn submit_and_watch_extrinsic_until_without_events<
```

Comment:
Can you move this function to the bottom of the definitions? I think `submit_and_watch_extrinsic_until` should be the most visible one.

Reply:
`submit_and_watch_extrinsic_until` belongs to a different trait.

Reply:
Oh Right. Since we're removing the `success` function, we should also rethink our Trait structure. I'd vote for moving the new function to `SubmitAndWatch` and remove `SubmitAndWatchUntilSuccess` completely once the deprecated functions are removed. I don't see the use case in having two different traits.

Reply:
I'd move the new `submit_and_watch_extrinsic_until` already now to the `SubmitAndWatch` trait. Otherwise, upon trait removal, users will have to change the trait import to keep using the function.  Or is there a downside to already implement the function in that trait?